doc/modules/ROOT/pages/design/transports.adoc

```diff
@@ -91,6 +91,8 @@ $ clj -R:nrepl -m nrepl.cmdline -t nrepl.transport/edn
 nREPL server started on port 63266 on host localhost - nrepl+edn://localhost:63266
 ----
 
-The main difference between the bencode transport and the EDN one is that instead of sending and receiving
-bencode dictionaries you'd be sending and receiving EDN maps. The structures of the messages is identical, only
-the data format changes.
+The main difference between the bencode transport and the EDN one is that instead of sending and receiving bencode dictionaries and lists, and being limited to integer and byte types, you'd be sending and receiving full EDN structures and types, including maps, vectors, lists, sets, strings, keywords and symbols. The structures of the messages is similar, only the data format changes.
+
+This may be useful in a couple of usecases: In some clients, including ClojureScript ones, where EDN is easier to handle than bencode. Furthermore, because this transport exposes more of the richer data types in the internals of nREPL and its middlewares, it may support more complex usecases.
+
+The introduction of the EDN transport does require more specification of the message format, whereby some vagueness was allowed for by the limitatioms of bencode. For example, `:op` values are strings, and `:status` is either a keyword, or a set of keywords. This may cause some surprised, unintended consequences. Thus tweaks to the EDN message format may be possible until the transport is declared mature. Documentation of this using `clojure.spec` is planned.
```

Comment:
Probably we should add a couple of code examples here and in the bencode section, so it'd be easier for people to understand the differences.

Reply:
I've added more details to the section now. Let me know if it's in the right direction? I feel the next step in documentation is to actually write a section on *messages*. We can probably more formally adopt the convention that in documentation, we express messages in EDN, and the messages we send/receive over the wire is either whatever `bencode/write-bencode` produces or the EDN as given.

Reply:
> I've added more details to the section now. Let me know if it's in the right direction? I feel the next step in documentation is to actually write a section on messages. We can probably more formally adopt the convention that in documentation, we express messages in EDN, and the messages we send/receive over the wire is either whatever bencode/write-bencode produces or the EDN as given.

You're definitely moving in the right direction with the latest round of updates. I like your idea. 

> I've done most of a spec for messages, which is sitting in another branch. I suspect that would serve as message specs more effectively than written documentation, given the intended audience of tool authors.

 I generally see the two approaches as complimentary, as it's kind of weird to just point people to some specs without providing them any explanations. But I agree that for people working a lot with nREPL probably the specs are going to be the single source of truth.

Reply:
I've updated the Message section to refer to the transport section for details, and elaborated a bit more on how the bencode transport works.